docs/_layouts/doc.html

```diff
@@ -47,7 +45,76 @@
 
   {% if page.url contains '/docs/terraform/' %}
   <script src="{{ site.url }}/js/hcl-to-json.js"></script>
+  <script src="{{ site.url }}/js/yaml.min.js"></script>
+  <script src="{{ site.url }}/js/highlight.pack.js"></script>
   <script src="{{ site.url }}/js/docs-tf-link-converter.js"></script>
+  <script>
+    $(function() {
+      $('pre code').each(function(){
+        var el = $(this),
+            yaml = null,
+            obj = null,
+            wasJSON = false,
+            block = el.parent().parent(),
+            tabs = "<nav><a href='#' class='yaml active'>yaml</a><a href='#' class='hcl'>hcl</a></nav>",
+            hcl = el.text();
+
+        try {
+
+          // dirty work around #0
+          // do nothing if single line
+          if (!/\n/.test(hcl)) throw 'nah!';
+
+          // dirty work around #1
+          // get rid of `...`
+          hcl = hcl.replace('...', '');
+
+          // dirty work around #2
+          // if starts with `$ ` or `export ` or `terraform `
+          if (/^\$|^export\s|^terraform\s/.test(hcl)) throw 'nah!';
+
+          // if it's already json
+          try {
+            obj = JSON.parse(hcl);
+            tabs = "<nav><a href='#' class='yaml active'>yaml</a><a href='#' class='json'>json</a></nav>";
+          } catch(e) {
+            obj = hcltojson(hcl);
```

Comment:
it would be better to take this `hcltojson(hcl)` part in to a `try..catch` block.